### ISSUE TITLE
Seperate folders and connections in connection manager

### ISSUE THREAD

**Issue by andreashager** [HeidiSQL/HeidiSQL#1828]
**Is your feature request related to a problem? Please describe.**
You can sort the connection manager by name but if you have folders and connections in you connection manager the connections and sorted inside the folders.

**Describe the solution you'd like**
I would prefer this sorting:

- First all folders sorted by connection name
- After the folders all connections sorted by connection name

Because if you sort it by for example last connection the folders are also before the connections 🙂

**Describe alternatives you've considered**
Maybe there could be an alternative column that have exactly this behavior like "Connection name but seperate folders and connetions".

**Additional context**
The current status:
![Connection-Manager](https://github.com/HeidiSQL/HeidiSQL/assets/3351175/b5acb2b3-4bf7-4ecf-88e6-5d19a6576375)

**Comment by ansgarbecker** [comment]
Yes I think sorting the folders before the connections is a good idea. Users may know that behaviour from the Windows Explorer.

**Comment by andreashager** [comment]
Is there already any progress on this or dou you know when you plan to implement this feature? 😊 My Folder tree is growing bigger and bigger and getting more confusing 😕 Thanks!

**Comment by ansgarbecker** [comment]
Just added that feature. "Folders at top" is now an item in the context menu. Click on it to disable and you get the behaviour like it was up to now. It's enabled by default, as I think this is reasonable for all.

<img width="505" height="370" alt="Image" src="https://github.com/user-attachments/assets/17c6ae5a-4cc6-487e-b91f-6a6cbce765c5" />

**Comment by andreashager** [comment]
Awesome, thank you 🥳 I tested it, works like a charm, now my view is clean again 🥳I will send you a donation 😊 

The only "weird" behavior that I have now is that German umlauts like Ö gets sorted after the folder list:
- Folder 1
- Oh a nice folder
- Thats another folder
- Ö folder

This sorting is the same behavior as before but I think it would be a nice to have that the folder with Ö is sorted after O character - should I create another issue for this?

**Comment by ansgarbecker** [comment]
Ah, thanks for the hint, that was an issue not only with the folders, but in any tree or list, for instance in the database tab and the left tree. I just fixed that by telling `SysUtils.CompareText` to use the locale.

But I see empty folders are sorted at the very end, so we're not yet finished here. Or is that just happening here?

**Comment by ansgarbecker** [comment]
Empty folders now sorted like non-empty ones. New build just created.

**Comment by andreashager** [comment]
> But I see empty folders are sorted at the very end, so we're not yet finished here. Or is that just happening here?
Nope it was not only you, I also had this problem 😓 

Great! 🥳 Now everything works like a charm 🥳 
Thanks for you quick solution 😊
